helpers/withOffline.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react'
+/**
+ * Provide a `isOffline` props
+ * @param {Component} WrappedComponent
+ */
+const withOffline = WrappedComponent =>
+  class OfflineStateHOC extends Component {
+    state = {
+      isOnline: window.navigator.onLine
+    }
+
+    componentDidMount() {
+      window.addEventListener('online', this.setOnline)
+      window.addEventListener('offline', this.setOffline)
+    }
+
+    componentWillUnmount() {
+      window.removeEventListener('online', this.setOnline)
+      window.removeEventListener('offline', this.setOffline)
+    }
+
+    setOnline = () => this.setState({ isOnline: true })
+
+    setOffline = () => this.setState({ isOnline: false })
+
+    render() {
+      const { isOnline } = this.state
+
+      return <WrappedComponent isOffline={!isOnline} {...this.props} />
```

Comment:
Precisely why i created isOffline instead of passing isOnline. Since generally I'm doing some special stuff when we're offline, I prefer to use in my app : 

```
if(isOffline) {
alert('no connection')
}
```

... instead of 
```
if(!isOnline) alert('no connection')
```